### ISSUE TITLE
Dataset preparation

### ISSUE THREAD

**Issue by robrechtme** [shtoshni92/g2p#4]
Hi!

I'm trying to replicate the results from your [paper](https://arxiv.org/pdf/1610.06540v1.pdf) for the CMUDict dataset, but I can't get it above 44% WER 12% PER.  
You mention the [cmusphinx](https://raw.githubusercontent.com/cmusphinx/cmudict/master/cmudict.dict) version, but did you use this dataset unprocessed? In the paper it seems like the stress suffices are removed  from the vowels, therefore making the task much easier.

Thanks!

**Comment by shtoshni** [comment]
Yes stress is removed and that indeed makes the task easier. This was done in line with previous work.  In retrospect, we should've stated this in the paper. Thanks for bringing this up.

**Comment by robrechtme** [comment]
There is a mention though that you used the same setting as in [Chen et al.](http://joneschen.org/spigdog/papers/a014d.pdf), which strips stress and  removes any acronyms/abbreviations from de dataset. I looked over it at first, my bad.  

Are your results by any chance comparable to the ones from [cmusphinx/g2p-seq2seq](https://github.com/cmusphinx/g2p-seq2seq)? It uses a Transformer model which, according to my intuition, would yield better results.

**Comment by shtoshni** [comment]
Interesting, I wasn't aware of a Tranformer implementation for G2P. Regarding the performance, I think the reported transformer numbers are indeed better than my numbers -- I checked _CMUdict PRONASYL 2007_, and it matches my CMUDict setup.

**Comment by shtoshni** [comment]
Actually, I'm not very sure. So this paper is almost 3 years old and I reran the CMU setup after upgrading tensorflow (to 1.5). My current setup actually gave much better numbers than paper -- almost 1 percent absolute improvement bringing the performance close to 20.7 [(Link).](https://github.com/shtoshni92/g2p/issues/3). So I believe the reported performance of 20.6 by CMUSphinx is within the standard variance. In fact, I'm pretty sure that training a little longer or changing the learning rate schedule can easily accomplish this. Anyways, I'm not sure that for a problem like G2P, Transformer can get you significant gains over LSTMs. These are very short sequences which both models should be capable of handling.

**Comment by robrechtme** [comment]
The main reason would be training speed, I think, but LSTMs on CMUDict are sill doable. Anyway, I will experiment with both models as a part of my thesis and I'll share my results.

**Comment by shtoshni** [comment]
So on a single Titan X GPU, I was able to finish CMUDict experiments within 3-4 hrs. You're right that Transformers can give a speed edge. Looking fwd to hearing about your results.
